editor/src/clj/editor/resource.clj

```diff
@@ -52,7 +52,17 @@
   (openable? [this])
   (editable? [this]))
 
-(def resource? (partial satisfies? Resource))
+(def ^{:arglists '([x])} resource?
+  (let [cache (atom {})
+        not-found (Object.)]
+    (fn resource? [x]
+      (let [c (class x)
+            ret (@cache c not-found)]
+        (if (identical? not-found ret)
+          (let [ret (satisfies? Resource x)]
+            (swap! cache assoc c ret)
+            ret)
+          ret)))))
```

Comment:
Making resource check much faster by using a `class->result` cache

Reply:
But `extends?` is supposed to check if the class implements a particular protocol? 
https://clojuredocs.org/clojure.core/extends_q

Are you perhaps thinking of some other function? 🤔

Reply:
It looks like `extends?` has a more narrow scope and does not walk the inheritance chain, which can lead to false negatives. Ah, it would be so much simpler if `Resource` was an interface...